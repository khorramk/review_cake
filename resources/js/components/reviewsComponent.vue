<template>
    <div class="row justify-content-center reviews-container">
        <div class="col-md-8 reviews-container__content">
            <div class="card reviews-container__content__review-card">
                <div class="card-header reviews-card__header">user review</div>
                <div class="card-body reviews-card__body" v-for="review in reviewsTest" v-bind:key="review.id" v-show="review.reviews">
                    <div class="card reviews-card__single-review__wrapper">
                        <div class="card-body reviews-card__single-review">
                                {{ review.reviews }}
                                <a  class="review-card__single-review_add-comment" style="float:right;  color:blue;  margin-left:10px;" href="/comments/create">ðŸ—¨</a>
                                <a class="review-card__single-review_edit-review" style="float:right;  color:blue;  margin-left:10px;" :href="`/reviews/${review.id}/edit`" >ðŸ–‰</a>
                                <form  class="review-card__single-review_remove-review" style="float:right;  color:blue;  margin-left:10px;"   @submit="remove(review.id)">
                                    <button type="submit" style="color:teal; background:none; border:none; " >â¨¯</button>
                                </form>
                        </div>
                    </div>
                    {{review.id}}
                    <Comments :review-id="review.id"/>
                </div>
            </div>
            <a href="/reviews/create" style="background:yellow; width:100px; height:50px; border-radius:100%; float:right; z-index:999; right:0; position: absolute; bottom:0; text-align:center; font-size: 2em;" class="review-card__link">+</a>
        </div>
    </div>
</template>

<script >
import Axios from "axios";
import Comments from './Comments';

export default ({
    data() {
        return {
            reviewsTest: [],
            csrf: document.querySelector('meta[name="csrf-token"]').getAttribute('content')
            
        };
    },
    mounted(){
        Axios.get('/api/cake-component/review')
            .then((response)=> this.reviewsTest = response.data)
            .catch((err)=> console.log(err))
            .then((resp)=> console.log(resp));
             
    },
    methods: {
        remove(id){
            console.log(id);
            
           Axios.delete(`/reviews/${id}`);       
        } 
    },
    components:{
        Comments
    }
});
</script>
