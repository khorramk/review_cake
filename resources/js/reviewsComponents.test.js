//test.js
import Vue from 'vue';
import reviewsComponent from './components/reviewsComponent';
import axios from 'axios';
import VueRouter from 'vue-router';

test('review component to havea mount option', () => {
  expect(typeof reviewsComponent.mounted).toBe('function');
});

test('check the reviews list data type', () => {
  expect(typeof reviewsComponent.data).toBe('function')
  const defualtData = reviewsComponent.data();
  expect(typeof defualtData.reviewsList).toBe('object');
 });


 test('sets the reviews list data after mount', () => {
   const vm = new Vue(reviewsComponent);
   return vm.$nextTick().then(()=> {
     
     expect(vm.reviewsList).not.toHaveLength(0);
   })
 });

 it('renders link', () => {
   const vm = new Vue(reviewsComponent);
   vm.$nextTick().then(()=> {

    expect(vm).toMatchSnapshot();
   }).catch().finally();
 });