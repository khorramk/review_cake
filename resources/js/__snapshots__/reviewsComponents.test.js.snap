// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "components": Object {
      "Reviews": Object {
        "components": Object {
          "CommentsComponent": Object {
            "data": [Function],
            "methods": Object {
              "submit": [Function],
            },
            "props": Object {
              "comment": Object {
                "type": [Function],
              },
              "reviewId": Object {
                "type": [Function],
              },
            },
            "render": [Function],
            "staticRenderFns": Array [],
          },
        },
        "data": [Function],
        "methods": Object {
          "remove": [Function],
        },
        "props": Object {
          "comments": Object {
            "default": [Function],
            "type": [Function],
          },
          "review": Object {
            "default": [Function],
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
    },
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "mounted": Array [
      [Function],
    ],
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "reviewsList": Array [
      Object {},
    ],
  },
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "components": Object {
        "Reviews": Object {
          "components": Object {
            "CommentsComponent": Object {
              "data": [Function],
              "methods": Object {
                "submit": [Function],
              },
              "props": Object {
                "comment": Object {
                  "type": [Function],
                },
                "reviewId": Object {
                  "type": [Function],
                },
              },
              "render": [Function],
              "staticRenderFns": Array [],
            },
          },
          "data": [Function],
          "methods": Object {
            "remove": [Function],
          },
          "props": Object {
            "comments": Object {
              "default": [Function],
              "type": [Function],
            },
            "review": Object {
              "default": [Function],
              "type": [Function],
            },
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
      },
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "mounted": Array [
        [Function],
      ],
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "reviewsList": Array [
        Object {},
      ],
    },
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 1,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "reviewsList": Array [
      Object {},
    ],
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 1,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "reviewsList": Array [
    Object {},
  ],
}
`;
